src/Compilers/CSharp/Portable/CodeGen/EmitArrayInitializer.cs

```diff
@@ -497,14 +522,27 @@ private bool TryEmitReadonlySpanAsBlobWrapper(NamedTypeSymbol spanType, BoundExp
             {
                 if (IsPeVerifyCompatEnabled())
                 {
+                    // We can't emit code that could cause PEVerify to warn.
                     return false;
                 }
 
                 if (createSpan is not null)
                 {
+                    if (lengthForConstructor != elementCount)
+                    {
+                        // We need to use RuntimeHelpers.CreateSpan, but the code has requested a subset of the elements.
+                        // That means the code is something like `new ReadOnlySpan<char>(new[] { 'a', 'b', 'c' }, 1, 2)`
+                        // rather than `new ReadOnlySpan<char>(new[] { 'b', 'c' })`.  If such a pattern is found to be
+                        // common, this could be augmented to accomodate it.  For now, we just return false to fail
+                        // to optimize this case.
+                        return false;
```

Comment:
> the blob will just contain the subarray?

Without additional work, the blob will contain everything and either that would need to be changed to just be the subset or the resulting span would need to be sliced. "Why not support this case". is I don't think such a construction will be at all common.

Reply:
Can you point to any existing real-world case where someone wrote this?

Reply:
> Without additional work, the blob will contain everything

That could be easily changed so that the blob would only contain the sliced array. I understand that this might not be super common, but my concern is that *since this is supported for the byte case*, my assumption would be that it would be supported for the newly supported types as well. It might be better to be consistent.

Reply:
Ah I see. Might be worth adding a comment saying that the case when `lengthForConstructor != elementCount` is only necessary for utf-8 literals, which is only needed for bytes.